app/controllers/users_controller.rb

```diff
@@ -17,13 +25,52 @@ def create
   end
 
   def show
-    return if (@user = User.find_by id: params[:id])
+    @user = User.find_by id: params[:id]
+    return if @user.present?
     flash[:danger] = t "danger"
   end
 
+  def edit
+    @user = User.find params[:id]
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update_attributes(user_params)
+      flash[:success] = t "profile_updated"
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t "user_deleted"
+    redirect_to users_url
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password, :password_confirmation
   end
+
+  def logged_in_user
+    unless logged_in?
```

Comment:
```suggestion
    return if logged_in?
```
xong xóa end ở dưới đi c nhé